tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -1200,6 +1343,7 @@ void CGMSHLSLRuntime::AddHLSLFunctionInfo(Function *F, const FunctionDecl *FD) {
       QualType recordTy = MD->getASTContext().getRecordType(RD);
       llvm::Type *Ty = CGM.getTypes().ConvertType(recordTy);
       AddResourceMetadata(recordTy, Ty);
+      AddHandleAnnotation(recordTy, Ty);
```

Comment:
I think both AddResourceMetadata and AddHandleAnnotation can't be determined here, and need to be filled in when a method is called based on the resource declaration (global, local, param, struct field) that has all the properties.